### PR TITLE
feat: introducing a swipeable way to toggle the sidebar

### PR DESCRIPTION
**Pull request by rebelchris**
Introducing the swipe-able hook to determine if we should open or close the sidebar on mobile.

Getting weird issues saying the hook usage is wrong, but not sure why.
This seems like a valid way of using this hook?

### CODE REVIEW

File: packages/shared/src/components/MainLayout.tsx [hunk 1]
```diff
@@ -89,6 +90,19 @@ export default function MainLayout({
...
+        setOpenMobileSidebar(false);
+      }
+    },
+    onSwipedRight: () => {
+      if (!sidebarRendered && !openMobileSidebar) {
+        setOpenMobileSidebar(true);
+      }
+    },
+    preventDefaultTouchmoveEvent: true,
+  });
```
Comment (idoshamun) [inline packages/shared/src/components/MainLayout.tsx:105]: This calls for a custom hook 😉

File: packages/shared/package.json [hunk 2]
```diff
@@ -112,6 +112,7 @@
     "lazysizes": "^5.3.2",
     "linkifyjs": "^2.1.9",
     "lodash.clonedeep": "^4.5.0",
-    "node-fetch": "^2.6.6"
+    "node-fetch": "^2.6.6",
+    "react-swipeable": "^6.2.0"
```
Comment (idoshamun) [inline packages/shared/package.json:116]: What's the package size? Have you checked if there are any lighter alternatives?

**Review by idoshamun — CHANGES_REQUESTED** [review]
Looks

File: packages/shared/package.json (hunk 2 above)
Comment (rebelchris) [reply packages/shared/package.json:116]: Unpacked Size
153 kB

Can have a look what else is out there :)

**Review by sshanzel — APPROVED** [review]
As neat as it can get! Just the hook that Ido have mentioned then I think this should be good to go 🚢

File: packages/shared/src/components/MainLayout.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/components/MainLayout.tsx:105]: Even with a custom hook i'm getting the error that the hook usage is wrong.
Do you have any idea why this occurs?

File: packages/shared/package.json (hunk 2 above)
Comment (idoshamun) [reply packages/shared/package.json:116]: Please fix the build issue so we can see how much payload it adds to our bundles

Comment (rebelchris) [reply packages/shared/package.json:116]: @idoshamun Having some issues with React not liking this hook implementation.
Do you have any guidance on where to look for this issue?

File: packages/shared/src/hooks/useSwipeableSidebar.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      if (!sidebarRendered && !openMobileSidebar) {
+        setOpenMobileSidebar(true);
+      }
+    },
+    preventDefaultTouchmoveEvent: true,
+  });
+
+  return useMemo(() => {
+    return {
+      ...handlers,
```
Comment (idoshamun) [inline packages/shared/src/hooks/useSwipeableSidebar.ts:30]:
```suggestion
      handlers,
```

**Review by idoshamun — CHANGES_REQUESTED** [review]
This should do it

**Review by idoshamun — APPROVED** [review]
Yay!

**Pull request merged**